assembl/graphql/schema.py

```diff
@@ -1440,6 +1476,35 @@ def mutate(root, args, context, info):
             post.body.add_value(body, original_body_entry.locale_code)
             changed = True
 
+            original_attachments = post.attachments
+            if original_attachments:
+                original_attachments_doc_ids = [str(a.document_id) for a in original_attachments]
+                attachments = args.get('attachments', [])
+                for document_id in attachments:
+                    if document_id not in original_attachments_doc_ids:
+                        document = models.Document.get(document_id)
+                        models.PostAttachment(
+                            document=document,
+                            discussion=discussion,
+                            creator_id=context.authenticated_userid,
+                            post=post,
+                            title=document.title,
+                            attachmentPurpose="EMBED_ATTACHMENT"
+                        )
+
+                # delete attachments that has been removed
+                documents_to_delete = set(original_attachments_doc_ids) - set(attachments)
+                for document_id in documents_to_delete:
+                    with cls.default_db.no_autoflush:
+                        document = models.Document.get(document_id)
+                        post_attachment = post.db.query(
+                            models.PostAttachment
+                        ).filter_by(
+                            discussion_id=discussion_id, post_id=post_id, document_id=document_id
+                            ).first()
+                        post.db.delete(document)
+                        post.attachments.remove(post_attachment)
+
```

Comment:
yes @cedricmessiant , you should flush at the end of a mutation. It may not be necessary for real because there is a commit at the end of the request, but it's not necessary the case in the tests and it can give errors in tests that are hard to debug.

Reply:
done